src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetActivityResourceIntTest.kt

```diff
@@ -561,17 +548,332 @@ class GetActivityResourceIntTest : IntegrationTestBase() {
+          offender {
+            booking(livingUnitId = -3009) {
+              courseAllocation(
+                courseActivity = courseActivity,
+                startDate = "$yesterday",
+                endDate = "$tomorrow",
+                endReasonCode = "WDRAWN",
+                endComment = "Withdrawn",
+                suspended = true,
+              ) {
+                payBand(payBandCode = "1")
+              }
+            }
+          }
+        }
+
+        webTestClient.getActivityDetails()
+          .expectBody()
+          .jsonPath("allocations[0].startDate").isEqualTo("$yesterday")
+          .jsonPath("allocations[0].endDate").isEqualTo("$tomorrow")
+          .jsonPath("allocations[0].endReasonCode").isEqualTo("WDRAWN")
+          .jsonPath("allocations[0].endComment").isEqualTo("Withdrawn")
+          .jsonPath("allocations[0].suspended").isEqualTo(true)
+          .jsonPath("allocations[0].payBand").isEqualTo("1")
+          .jsonPath("allocations[0].livingUnitDescription").isEqualTo("BXI-A-1-016")
+      }
+
+      @Test
+      fun `should not include allocations in the wrong status`() {
+        nomisDataBuilder.build {
+          programService {
+            courseActivity = courseActivity()
+          }
+          offender(nomsId = "A1111AA") {
+            booking {
+              courseAllocation(
+                courseActivity = courseActivity,
+                programStatusCode = "ALLOC",
+              )
+            }
+          }
+          offender(nomsId = "A2222AA") {
+            booking {
+              courseAllocation(
+                courseActivity = courseActivity,
+                programStatusCode = "END",
+              )
+            }
+          }
+        }
+
+        webTestClient.getActivityDetails()
+          .expectBody()
+          .jsonPath("allocations[0].nomisId").isEqualTo("A1111AA")
+          .jsonPath("allocations[1]").doesNotExist()
+      }
+
+      @Test
+      fun `should not include allocations which are inactive by date `() {
+        nomisDataBuilder.build {
+          programService {
+            courseActivity = courseActivity()
+          }
+          offender(nomsId = "A1111AA") {
+            booking {
+              courseAllocation(
+                courseActivity = courseActivity,
+                startDate = "$yesterday",
+                endDate = null,
+              )
+            }
+          }
+          offender(nomsId = "A2222AA") {
+            booking {
+              courseAllocation(
+                courseActivity = courseActivity,
+                startDate = "$yesterday",
+                endDate = "$yesterday",
+              )
+            }
+          }
+        }
+
+        webTestClient.getActivityDetails()
+          .expectBody()
+          .jsonPath("allocations[0].nomisId").isEqualTo("A1111AA")
+          .jsonPath("allocations[1]").doesNotExist()
+      }
+
+      @Test
+      fun `should not include prisoners who are OUT`() {
```

Comment:
are you sure? if they are OUT won't they longer have an active allocation anyway?
Are there in weird scenarios where they are out at a court in another city and we want to keep their allocation?
Are you 100% certain this correct feels scary

Reply:
I think it's just a badly named test. The code actually checks the `ACTIVE_FLAG`, not the `IN_OUT_STATUS`. I'll rename the test in my next PR.